### PR TITLE
[MMCA-4874] - ACC44 - Create connector for Cash Account Transaction Search

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - ACC44 connector and related code changes for making a ACC44 request

Below has been done as part of this PR

- add ACC44 request and response schema
- add Request and Response models
- add ACC44 connector
- add new test classes
- add new controller and service for ACC44
- add new route
- add new config entry for ACC44 endpoint
- refactoring

### CODE REVIEW

**Review by wg-hmrc — APPROVED** [review]

File: app/connectors/Acc44Connector.scala [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.AppConfig
+import config.MetaConfig.Platform.MDTP
+import models.requests.{CashAccountTransactionSearchRequest, CashAccountTransactionSearchRequestContainer,
```
Comment (AdamCoops) [inline app/connectors/Acc44Connector.scala:21]: give how long this is I'd either break it up so like

import abc.123.{
  itemOne,
  itemTwo}
  
  
 or just do ._ imo

File: app/connectors/Acc44Connector.scala [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  private def postValidRequest(cashAccTransSearchRequestContainer: CashAccountTransactionSearchRequestContainer):
+  Future[Either[ErrorDetail, CashAccountTransactionSearchResponseContainer]] = {
+
+    metricsReporterService.withResponseTimeLogging("hods.post.cash-account-transaction-search") {
+      httpClient.POST[CashAccountTransactionSearchRequestContainer, HttpResponse](
+        appConfig.acc44CashTransactionSearchEndpoint,
+        cashAccTransSearchRequestContainer,
+        headers = headers.headers(appConfig.acc44BearerToken, None)
+      )(implicitly, implicitly, HeaderCarrier(), implicitly).map { res =>
+        res.status match {
```
Comment (AdamCoops) [inline app/connectors/Acc44Connector.scala:90]: I think you want this with a space between it and 89 for readability?

File: app/connectors/Acc44Connector.scala (hunk 1 above)
Comment (tyagikumarvineet) [reply app/connectors/Acc44Connector.scala:21]: Adam, its okay for imports like this as its within 120 characters and on two lines. Cannot import all the other models that are not required as this adds the unnecessary overhead for the class and memory.

File: app/connectors/Acc44Connector.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply app/connectors/Acc44Connector.scala:91]: Done

**Review by AdamCoops — APPROVED** [review]

**Review by wg-hmrc — APPROVED** [review]

**Pull request merged**